invokeai/frontend/web/src/features/nodes/components/flow/Flow.tsx

```diff
@@ -149,10 +151,26 @@ export const Flow = () => {
     dispatch(contextMenusClosed());
   }, [dispatch]);
 
-  const onInit: OnInit = useCallback((flow) => {
-    $flow.set(flow);
-    flow.fitView();
-  }, []);
+  const onInit: OnInit = useCallback(
+    (flow) => {
+      $flow.set(flow);
+      flow.fitView();
+      flowWrapper.current = document.getElementById('workflow-editor');
+    },
```

Comment:
For this particular situation, we don't need the cursor position in redux. We can store it in a ref in this component and pass it as a payload with the paste action.

This avoids constant state updates in the redux store, which can cause performance issues for consumers of the nodes state. You can see this if you enable `Highlight updates when components render.` in the react devtools and move the cursor. Admittedly, this indicates we need to fix some components somewhere.

Refs maintain a single reference in memory for the lifetime of the component and therefore don't trigger side effects (like virtual DOM reconciliation or re-renders) when their value changes. When you don't need to actually react to a value changing, a ref is often a good place to put the value.

Reply:
As a side-note, if we did want to store this in redux, we'd want to exclude it from being saved in `localStorage`, by adding it to `nodesPersistDenylist.ts`.